src/react/useQueries.ts

```diff
@@ -2,10 +2,29 @@ import React from 'react'
 
 import { notifyManager } from '../core/notifyManager'
 import { QueriesObserver } from '../core/queriesObserver'
+import { InitialDataFunction, PlaceholderDataFunction } from '../core/types'
 import { useQueryClient } from './QueryClientProvider'
 import { UseQueryOptions, UseQueryResult } from './types'
 
-export function useQueries(queries: UseQueryOptions[]): UseQueryResult[] {
+type InferredQueryOptions<TQueryFnData> = Omit<
+  UseQueryOptions<TQueryFnData, unknown, TQueryFnData>,
+  'select' | 'initialData' | 'placeholderData'
+> & {
+  select?: (data: TQueryFnData) => unknown
+  initialData?: TQueryFnData extends infer TInitialData
+    ? TInitialData | InitialDataFunction<TInitialData>
+    : never
+  placeholderData?: TQueryFnData extends infer TPlaceholderData
+  ? TPlaceholderData | PlaceholderDataFunction<TPlaceholderData>
+  : never
```

Comment:
Using conditional type with inference, to check that the type of `initialData` and `placeholderData` corresponds to the same type as returned from the `queryFn`.

Without this, if you added one of these properties with a different type from the `queryFn`, TS would merge the type definitions into a union 😢 

For example, the below would make TS think that `TQueryFnData` was `string | number` instead of just `string`

```typescript
useQueries([
  {
    queryKey: key3,
    queryFn: () => 'string',
    initialData: 2,
  },
])
```

Reply:
Hmm but why would select return unknown? We don’t get type inference of the returned data then depending on select, right?

Reply:
aha - that was a knowledge gap on my part; I misunderstood `select` completely.. I think I get it now though (its return type should take precedence over `queryFn` when it's defined?)

Reply:
What you return from select overrides what the queryFn usually returns from useQueries, but keep in mind that placeholderData and initialData still need to be of type TQueryFnData, because they also run through select

Reply:
(latest commit should handle this now 🤞 )